front/app/containers/Admin/pagesAndMenu/EditHomepage/HeroBanner/index.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { useState } from 'react';
+
+// components
+import GenericHeroBannerForm, {
+  HeroBannerInputSettings,
+} from '../../containers/GenericHeroBannerForm';
+import {
+  pagesAndMenuBreadcrumb,
+  homeBreadcrumb,
+} from 'containers/Admin/pagesAndMenu/breadcrumbs';
+
+// resources
+import useHomepageSettings from 'hooks/useHomepageSettings';
+import {
+  IHomepageSettingsAttributes,
+  updateHomepageSettings,
+} from 'services/homepageSettings';
+
+// utils
+import { isNilOrError } from 'utils/helperUtils';
+import { forOwn, isEqual } from 'lodash-es';
+
+// i18n
+// change
+import messages from '../../containers/GenericHeroBannerForm/messages';
+import { InjectedIntlProps } from 'react-intl';
+import { injectIntl } from 'utils/cl-intl';
+import { ISubmitState } from 'components/admin/SubmitWrapper';
+
+const EditHomepageHeroBannerForm = ({
+  intl: { formatMessage },
+}: InjectedIntlProps) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [formStatus, setFormStatus] = useState<ISubmitState>('disabled');
+
+  const homepageSettings = useHomepageSettings();
+
+  if (isNilOrError(homepageSettings)) {
+    return null;
+  }
+
+  const { attributes } = homepageSettings;
+
+  const handleSave = async (newSettings: HeroBannerInputSettings) => {
+    if (!newSettings) return;
+
+    // some names in the generic form are different from the property on Homepage
+    const propsMappedToHomepageSettingsNames: Partial<IHomepageSettingsAttributes> =
+      {
+        banner_signed_out_header_overlay_opacity:
+          newSettings.banner_overlay_opacity,
+        banner_signed_out_header_overlay_color:
+          newSettings.banner_overlay_color,
+        banner_signed_out_header_multiloc: newSettings.banner_header_multiloc,
+        banner_signed_out_subheader_multiloc:
+          newSettings.banner_subheader_multiloc,
+        // the rest are the same as used in the form
+        ...newSettings,
+      };
+
+    // only update the page settings if they have changed
+    const diffedValues = {};
+    forOwn(propsMappedToHomepageSettingsNames, (value, key) => {
+      if (!isEqual(value, attributes[key])) {
+        diffedValues[key] = value;
+      }
+    });
+
+    setIsLoading(true);
+    setFormStatus('disabled');
+    try {
+      await updateHomepageSettings(diffedValues);
+      setIsLoading(false);
+      setFormStatus('success');
+    } catch (error) {
+      setIsLoading(false);
+      setFormStatus('error');
+    }
+  };
+
+  // this could probably be done smarter without enumerating everything, but
```

Comment:
See the comment here, any ideas about how to clean this up are appreciated. Mainly wanted to get the functionality working first